x-pack/plugins/apm/server/lib/fleet/register_fleet_policy_callbacks.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { APMPlugin, APMRouteHandlerResources } from '../..';
+import { listConfigurations } from '../settings/agent_configuration/list_configurations';
+import { setupRequest } from '../helpers/setup_request';
+import { APMPluginStartDependencies } from '../../types';
+import { ExternalCallback } from '../../../../fleet/server';
+
+export async function registerFleetPolicyCallbacks({
+  plugins,
+  ruleDataClient,
+  config,
+  logger,
+}: {
+  plugins: APMRouteHandlerResources['plugins'];
+  ruleDataClient: APMRouteHandlerResources['ruleDataClient'];
+  config: NonNullable<APMPlugin['currentConfig']>;
+  logger: NonNullable<APMPlugin['logger']>;
+}) {
+  if (!plugins.fleet) {
+    return;
+  }
+  const fleetPluginStart = await plugins.fleet.start();
+
+  registerAgentConfigExternalCallback({
+    fleetPluginStart,
+    callbackName: 'packagePolicyCreate',
+    plugins,
+    ruleDataClient,
+    config,
+    logger,
+  });
+  registerAgentConfigExternalCallback({
+    fleetPluginStart,
+    callbackName: 'packagePolicyUpdate',
+    plugins,
+    ruleDataClient,
+    config,
+    logger,
+  });
+}
+
+type ExternalCallbackParams = Parameters<ExternalCallback[1]>;
+type PackagePolicy = ExternalCallbackParams[0];
+type Context = ExternalCallbackParams[1];
+type Request = ExternalCallbackParams[2];
+
+function registerAgentConfigExternalCallback({
+  fleetPluginStart,
+  callbackName,
+  plugins,
+  ruleDataClient,
+  config,
+  logger,
+}: {
+  fleetPluginStart: NonNullable<APMPluginStartDependencies['fleet']>;
+  callbackName: ExternalCallback[0];
+  plugins: APMRouteHandlerResources['plugins'];
+  ruleDataClient: APMRouteHandlerResources['ruleDataClient'];
+  config: NonNullable<APMPlugin['currentConfig']>;
+  logger: NonNullable<APMPlugin['logger']>;
+}) {
+  const callbackFn: ExternalCallback[1] = async (
+    packagePolicy: PackagePolicy,
+    context: Context,
+    request: Request
+  ) => {
+    if (packagePolicy.package?.name !== 'apm') {
+      return packagePolicy;
+    }
+    const setup = await setupRequest({
+      context: context as any,
+      params: { query: { _inspect: false } },
+      core: null as any,
+      plugins,
+      request,
+      config,
+      logger,
+      ruleDataClient,
+    });
+    const configurations = await listConfigurations({ setup });
+    const agentConfigValue = configurations.map((configuration) => ({
+      service: configuration.service,
+      settings: configuration.settings,
```

Comment:
```suggestion
      config: configuration.settings,
```

Here's how things look in apm-server currentlly: https://github.com/elastic/apm-server/blob/91645bde2d96a79b943c90253d62da3861ab2997/apm-server.yml#L275-L297

```
  #---------------------------- APM Server - Agent Configuration ----------------------------
  # Directly specify agent configuration. If `agent_config` is set, agent
  # configuration under `kibana` will be ignored.
  # An agent's incoming configuration request will be matched to an
  # agent_config with the following precedence:
  # - service.name and service.environment match an agent_config
  # - service.name matches an agent_config, service.environment == ""
  # - service.environment matches an agent_config, service.name == ""
  # - an agent_config without a name or environment set
  # An empty result is is returned if no matching result is found.
  # agent_config:
  # - service.name: ten_percent
  #   service.environment: production
  #   config:
  #     capture_body: off
  #     capture_body: true
  #     log_level: info
  #     recording: true
  #     transaction_sample_rate: 0.1
  # - service.name: frontend
  #   agent.name: rum-js
  #   config:
  #     transaction_sample_rate: 0.1
```

We could alternatively change the apm-server implementation if you/anyone feels strongly about calling this "settings".

Reply:
I'm ok changing it to conform with apm-server's implementation. thanks!